.gitlab-ci.yml

```diff
@@ -2585,6 +2586,30 @@ deploy_puppy:
     - export PACKAGE_VERSION=$(inv agent.version --url-safe --major-version 7)
     - aws s3 cp --region us-east-1 ./agent $S3_DSD6_URI/puppy/agent-$PACKAGE_VERSION --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
 
+# deploy cloudfoundry bosh zip to the staging bucket
+deploy_cloudfoundry_bosh_agent_zip:
+  <<: *run_when_triggered_on_tag_7
+  stage: deploy7
+  needs: [ "windows_dsd_msi_x64"]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-builders/deploy:$DATADOG_AGENT_BUILDERS
+  before_script:
+    - ls $OMNIBUS_PACKAGE_DIR
+  tags: [ "runner:main", "size:large" ]
+  script:
+    - $S3_CP_CMD --recursive --exclude "*" --include "datadog-agent-7.*.zip" $OMNIBUS_PACKAGE_DIR s3://$S3_DSD6_BASE/windows/agent7/bosh/ --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers full=id=3a6e02b08553fd157ae3fb918945dd1eaae5a1aa818940381ef07a430cf25732
```

Comment:
I don't have the full context in mind, but is the package in the zip file only meant to work on bosh?

In terms of naming, to me it's more future-proof to push the artifacts to a location that's named after what the artifacts are/do, instead of what the artifacts are used for later in the pipeline. Rationale is that we may use the same artifacts for other purposes in the future, but what they are/do shouldn't change significantly. Note that if there's anything in these artifacts that's very specific to bosh, then the current name makes sense.

Also, why is the artifact named `datadog-agent` if it only contains dogstatsd?

Reply:
Thanks for the review. This zip includes the full agent, not just dogstatsd. As the dsd6-staging bucket holds things other than dogstatsd (like puppy) this seemed like an OK place to put it. I can move it if there's a better place. 

For the naming, @derekwbrown do you have any thoughts on a good name for this package since it could theoretically be used for more than bosh?

Reply:
To help clarify things:

The `cloudfoundry bosh zip` step is for a zip package that includes an already unpacked, full, datadog agent with an extra install executable to help setup things like the user, Maybe we can call this simply: `datadog-agent-windows-<version>.zip`
The `cloudfoundry buildpack zip` step is for a zip package that includes all the standalone binaries for windows, To Albert's point, Maybe we can call this: something like: `datadog-agent-windows-binaries-<version>.zip`
Open to other names though 😄

Reply:
Those names are fine for me. We can rename them on a separate PR, though, since it's out of the scope of this one.

Reply:
ok, I think I was confused because the job depends on a job named `windows_dsd_msi_x64`, so I assumed there it was a dogstatsd package. Is the dependency correct btw?

Anyway, these names sound fine to me as well, just want to make sure that the names make sense 👍

Reply:
🤦‍♂ That was indeed not the right job dependency, I just fixed it. While doing that, I updated the naming of the jobs, but as Albert mentioned, I'll leave the names of the artifacts in this PR.